storage/blob.go

```diff
@@ -438,6 +496,124 @@ func (b BlobStorageClient) ContainerExists(name string) (bool, error) {
+	}
+
+	if containerPermissions.AccessOptions.LeaseID != "" {
+		headers[leaseID] = containerPermissions.AccessOptions.LeaseID
+	}
+
+	// generate the XML for the SharedAccessSignature if required.
+	accessPolicyXML := b.generateAccessPolicy(containerPermissions.AccessPolicy)
+
+	var resp *storageResponse
+	var err error
+	if accessPolicyXML != "" {
+		headers["Content-Length"] = fmt.Sprintf("%v", len(accessPolicyXML))
+		resp, err = b.client.exec("PUT", uri, headers, strings.NewReader(accessPolicyXML))
+	} else {
+		resp, err = b.client.exec("PUT", uri, headers, nil)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if resp != nil {
+		defer resp.body.Close()
+
+		if resp.statusCode != http.StatusOK {
+			return errors.New("Unable to set permissions")
+		}
+	}
+	return nil
+}
+
+// GetContainerPermissions gets the container permissions as per https://msdn.microsoft.com/en-us/library/azure/dd179469.aspx
+// If timeout is 0 then it will not be passed to Azure
+// leaseID will only be passed to Azure if populated
+// Returns permissionResponse which is combined permissions and AccessPolicy
+func (b BlobStorageClient) GetContainerPermissions(container string, timeout int, leaseID string) (permissionResponse *ContainerAccessResponse, err error) {
+	params := url.Values{"restype": {"container"},
+		"comp": {"acl"}}
+
+	if timeout > 0 {
+		params.Add("timeout", strconv.Itoa(timeout))
+	}
+
+	uri := b.client.getEndpoint(blobServiceName, pathForContainer(container), params)
+	headers := b.client.getStandardHeaders()
+
+	if leaseID != "" {
+		headers[leaseID] = leaseID
+	}
+
+	resp, err := b.client.exec("GET", uri, headers, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// containerAccess. Blob, Container, empty
+	containerAccess := resp.headers.Get(http.CanonicalHeaderKey(ContainerAccessHeader))
+
+	defer resp.body.Close()
+	var out AccessPolicyResponse
+	err = xmlUnmarshal(resp.body, &out.SignedIdentifiersList)
+	if err != nil {
+		return nil, err
+	}
+
+	permissionResponse = &ContainerAccessResponse{}
+	permissionResponse.AccessPolicy = out.SignedIdentifiersList
+	permissionResponse.ContainerAccess = ContainerAccessType(containerAccess)
+
+	return permissionResponse, nil
+}
+
+// generateAccessPolicy generates the XML access policy used as the payload for SetContainerPermissions.
+// TODO(kpfaulkner) find better way to generate the XML. This is clunky.
+func (b BlobStorageClient) generateAccessPolicy(accessPolicy AccessPolicyDetails) string {
+
+	if accessPolicy.ID != "" {
+		s := `<?xml version="1.0" encoding="utf-8"?><SignedIdentifiers><SignedIdentifier>`
+		s += fmt.Sprintf("<Id>%s</Id>", accessPolicy.ID)
+		s += "<AccessPolicy>"
+
+		permissions := ""
+		if accessPolicy.CanRead {
+			permissions += "r"
+		}
+
+		if accessPolicy.CanWrite {
+			permissions += "w"
+		}
+
+		if accessPolicy.CanDelete {
+			permissions += "d"
+		}
+
+		s += fmt.Sprintf("<Start>%s</Start>", accessPolicy.StartTime.Format("2006-01-02T15:04:05Z"))
```

Comment:
Would one of this [consts](https://golang.org/pkg/time/#pkg-constants) work instead of `"2006-01-02T15:04:05Z"` ?